packages/app-check/src/internal-api.test.ts

```diff
@@ -281,11 +289,70 @@ describe('internal api', () => {
       );
       expect(token).to.deep.equal({ token: fakeRecaptchaAppCheckToken.token });
     });
+
+    it('reads any memory-cached debug token if in debug mode', async () => {
```

Comment:
I'm unclear what it is testing. Do you mean the memory cached appcheck token exchanged from a debug token?

I don't think we need a special test case for reading memory cached appcheck token in the debug mode since we don't treat it differently than in prod mode.

Reply:
Ditto for the other 2 new tests below.

Reply:
Ok, I guess it was just to make sure everything acts as expected and we don't regress if someone changes the order of the `if` checks in the future, but I guess we probably won't do that.